pkg/controllers/state/statenode.go

```diff
@@ -517,3 +518,31 @@ func RequireNoScheduleTaint(ctx context.Context, kubeClient client.Client, addTa
 	}
 	return multiErr
 }
+
+// ClearNodeClaimsCondition will remove the conditionType from the NodeClaim status of the provided statenodes
+func ClearNodeClaimsCondition(ctx context.Context, kubeClient client.Client, conditionType string, nodes ...*StateNode) (err error, requeue bool) {
+	return multierr.Combine(lo.Map(nodes, func(s *StateNode, _ int) error {
+		if !s.Initialized() || s.NodeClaim == nil {
+			return nil
+		}
+		nodeClaim := &v1.NodeClaim{}
+		err := kubeClient.Get(ctx, client.ObjectKeyFromObject(s.NodeClaim), nodeClaim)
+		if err != nil {
+			return err
+		}
+		if !nodeClaim.StatusConditions().IsTrue(conditionType) {
```

Comment:
Do we want to keep the status condition around if it's false? Probably easier just to clear then do a deep equals diff?